HousingManagementSystemApi.Tests/GatewaysTests/AddressesCosmosGatewayTests.cs

```diff
@@ -0,0 +1,97 @@
+        {
+            this.cosmosQueryHelperMock = new Mock<ICosmosAddressQueryHelper>();
+            this.systemUnderTest = new AddressesCosmosGateway(this.cosmosQueryHelperMock.Object);
+            this.feedIteratorMock = new Mock<FeedIterator<PropertyAddress>>();
+        }
+
+        [Fact]
+        public async void Test_Get_addresses_Not_Throws_Exception()
+        {
+            // Arrange
+            this.feedIteratorMock.Setup(_ => _.HasMoreResults).Returns(false);
+            this.cosmosQueryHelperMock.Setup(_ => _.GetItemQueryIterator<PropertyAddress>("NG21 9LQ"))
+                .Returns(this.feedIteratorMock.Object);
+
+            // Act
+            Func<Task> act = async () => await this.systemUnderTest.SearchByPostcode("NG21 9LQ");
+
+            // Assert
+            await act.Should().NotThrowAsync();
+        }
+
+        [Fact]
+        public async void Test_No_Addresses_Returned_When_Iterator_No_Results()
+        {
+            // Arrange
+            this.feedIteratorMock.Setup(_ => _.HasMoreResults).Returns(false);
+            this.cosmosQueryHelperMock.Setup(_ => _.GetItemQueryIterator<PropertyAddress>("NG21 9LQ"))
+                .Returns(this.feedIteratorMock.Object);
+
+            // Act
+            var results = await this.systemUnderTest.SearchByPostcode("NG21 9LQ");
+
+            // Assert
+            Assert.True(!results.Any());
+        }
+
+        [Fact]
+        public async void Test_Addresses_Returned_When_Iterator_Results()
+        {
+            // Arrange
+            this.feedIteratorMock.Setup(_ => _.HasMoreResults).Returns(true);
+            var addressList = new List<PropertyAddress>();
+            addressList.Add(new PropertyAddress() { CityName = "test", PostalCode = "NG21 9LQ" });
+
+            var feedResponseMock = new Mock<FeedResponse<PropertyAddress>>();
+            feedResponseMock.Setup(x => x.GetEnumerator()).Returns(addressList.GetEnumerator());
+
+
+            this.feedIteratorMock
+                .Setup(f => f.ReadNextAsync(It.IsAny<CancellationToken>()))
+                .ReturnsAsync(feedResponseMock.Object)
+                .Callback(() => this.feedIteratorMock
+                    .Setup(f => f.HasMoreResults)
+                    .Returns(false));
+
+            var containerMock = new Mock<Container>();
+            containerMock
+                .Setup(c => c.GetItemQueryIterator<PropertyAddress>(
+                    It.IsAny<QueryDefinition>(),
+                    It.IsAny<string>(),
+                    It.IsAny<QueryRequestOptions>()))
+                .Returns(this.feedIteratorMock.Object);
+
+            this.cosmosQueryHelperMock.Setup(_ => _.GetItemQueryIterator<PropertyAddress>("NG21 9LQ"))
+                .Returns(this.feedIteratorMock.Object);
+
+            // Act
+            var results = await this.systemUnderTest.SearchByPostcode("NG21 9LQ");
+
+            // Assert
+            Assert.True(results.Count() == 1);
```

Comment:
This assert wouldn't show an informative message on failure:
```
Expected: True
Actual:   False
```

An informative error message can be useful in understanding the reason for failure.
Using `Assert.Equal(1, results.Counts())` would give the following error message:
```
Expected: 1
Actual:   0
```
Using `Assert.Single(results)` gives the following error message:
```
The collection was expected to contain a single element, but it was empty.
```

The latter example is preferred, i.e.
```suggestion
            Assert.Single(results);
```

Reply:
Thanks, I didn't know about this.